### PR TITLE
Save and Reload Free/Solution Gas/Oil Cumulatives in Restart

### PR DESCRIPTION
**Pull request by bska**
This commit ensures that the field, group, and well level cumulative production curves of solution/free oil and gas are continuous in a restarted simulation run.

Thanks to Torbjørn Skille for identifying the appropriate locations in `XGRP`/`XWEL` for outputting these items.

### CODE REVIEW

**Comment by bska** [comment]
Here are a couple of comparisons between base runs of a slightly modified version of [`NOPINCH.DATA`](https://github.com/OPM/opm-tests/blob/570a0219b7efef7e07f4ec90240784a29d2fc13a/pinch/NOPINCH.DATA).  I disabled the tracers since tracer restart depends on OPM/opm-simulators#3708 and added more report/restart steps.

Summary Vector | Comparison | Detailed Comparison
------------- | ---------------- | ---------------
Field-level free oil cumulative production | FOPTF |
Field-level solution oil cumulative production | FOPTS | FOPTS-excerpt
Group-level free gas cumulative production | GGPTF | GGPTF-excerpt
Group-level solution gas cumulative production | GGPTS |
Well-level free gas cumulative production | WGPTF | WGPTF-excerpt
Well-level solution gas cumulative production | WGPTS |
Well-level free oil cumulative production | WOPTF |
Well-level solution oil cumulative production | WOPTS | WOPTS-excerpt

**Comment by bska** [comment]
Note that as this PR outputs new data items to `XGRP` and `XWEL` we will need to update the reference solutions of all existing regression tests to account for the new items.

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this update_data please

**Comment by bska** [comment]
jenkins build this opm-tests=686 please

**Comment by bska** [comment]
Resetting this back to draft state while I investigate the parallel regression failure.

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
jenkins build this opm-simulators=3675 please

**Comment by bska** [comment]
jenkins build this opm-simulators=3765 please

**Comment by bska** [comment]
With OPM/opm-simulators#3765 now merged into master there is no longer anything blocking this PR.  I'm marking it as "ready for review" and running a build check to verify that the only regression failures are the ones expected from having additional non-zero items in the restart files.

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
> I'm marking it as "ready for review" and running a build check to verify that the only regression failures are the ones expected from having additional non-zero items in the restart files.

I have reviewed the [regression failures](https://ci.opm-project.org/job/opm-common-PR-builder/4684/testReport/) of the build check and I'm confident that the failures are all due to outputting new items to the `XGRP` and `XWEL` arrays and nothing else.  I therefore deem it safe to update the reference solutions once the PR is approved.

**Review by joakim-hove — APPROVED** [review]

**Comment by joakim-hove** [comment]
jenkins build this update_data please

**Comment by joakim-hove** [comment]
jenkins build this opm-tests=709 please

**Pull request merged**
